src/app/modules/nodeRewardsProgram/nodeRewardsProgram.controller.js

```diff
@@ -1,21 +1,364 @@
+                }
+            });
+
+            // Build for account dropdown list
+            this.accounts = [...multisigAccounts, currentAccount];
+
+            this.selectedAccount = this.accounts[0];
+            this.balance = this.accounts[0].formatBalance;
+            this.formData.mainPublicKey = this.accounts[0].publicKey;
+        } else {
+            this.selectedAccount = currentAccount;
+            this.balance = currentAccount.formatBalance;
+            this.formData.mainPublicKey = currentAccount.publicKey;
+        }
+
+        this.balance = this.selectedAccount.formatBalance;
+        this.formData.mainPublicKey = this.selectedAccount.publicKey;
+        this.formData.isMultisig = this.selectedAccount.isMultisig || false;
+    }
+
+    /**
+     * Format absolute XEM balance to relative string
+     * @param {number} absoluteBalance
+     * @returns {string} Relative balance string
+     */
+    computeXEMBalance(absoluteBalance) {
+        return this._filter("fmtNemValue")(absoluteBalance || 0)[0] + "." + this._filter("fmtNemValue")(absoluteBalance || 0)[1];
+    }
+
+    /**
+     * Verify current selected tab
+     * @param {'status'|'enroll'|'payout'} tab
+     * @returns {boolean} boolean
+     */
+    isTabSelected(tab) {
+        return (this.selectedTab === tab);
+    }
+
+    /**
+     * Set tab
+     * @param {'status'|'enroll'|'payout'} tab
+     */
+    setTab(tab) {
+        this.selectedTab = tab;
+
+        switch (tab) {
+            case this.tab.status:
+                this.getStatus();
+                break;
+            case this.tab.enroll:
+                this.setEnroll();
+                break;
+            case this.tab.payout:
+                this.getPayoutHistory();
+                break;
+        }
+    }
+
+    /**
+     * Reset current tab to status
+     */
+    resetTab() {
+        this.setTab(this.tab.status);
+    }
+
+    /**
+     * Switch account if account contain multisig
+     */
+    onChangeAccount() {
+        this.balance = this.selectedAccount.formatBalance;
+        this.formData.isMultisig = this.selectedAccount.isMultisig || false;
+        this.formData.mainPublicKey = this.selectedAccount.publicKey;
+
+        this.resetTab();
+    }
+
+    /**
+     * Prepare the transaction
+     * @param {string} codewordHash
+     * @returns {object} transaction object
+     */
+    prepareTransaction(codewordHash) {
+        // Create a new object to not affect the view
+        let transferTransaction = nem.model.objects.get("transferTransaction");
+
+        // Set recipient
+        transferTransaction.recipient = this.formData.enrollAddress.toUpperCase().replace(/-/g, '');
+
+        // Set amount
+        transferTransaction.amount = 0
+
+        // Set empty mosaics
+        transferTransaction.mosaics = null;
+
+        // Set the message
+        transferTransaction.message = `enroll ${this.formData.domainOrIP} ${codewordHash ? codewordHash :'0'.repeat(64)}`;
```

Comment:
The reason I pre-defined `'0'.repeat(64)` is because I like to get the transaction fees correctly (final fees).

I will add another validation check for `codewordHash` before the user announces tx.